docs/spec_draft.md

```diff
@@ -874,6 +875,46 @@ implementation-defined value.
 
 [Back to Ops](#index-of-documented-ops)
 
+## stablehlo.reshape
+
+`stablehlo.reshape(operand) -> result`
+
+### Semantics
+
+Performs reshape of `operand` tensor to a `result` tensor. Conceptually, it
+collapses the dimensions of `operand` starting from slowest-varying
+dimension `0` to fastest-varying dimension `rank-1`, and then refines the
+flatenned vector into the shape of  `result`.
```

Comment:
This explanation uses some concepts, e.g. "collapse the dimension", "slowest-varying dimension", "fastest-varying dimension", "flattened vector", "refine into a shape", which haven't been defined above.

Strictly speaking, the "element-wise" concept in "performs element-wise <operation>" which we're using in the current draft is also not defined above, but its fairly intuitive already and easily searchable. However, the concepts used in this explanation are a mix of somewhat searchable things (collapsing a dimension) and outright implementation details (flattening into a vector, refining into a shape).

Let's define op semantics from first principles, using only the definitions in this spec and college maths. Before proceeding with that, let's first expand our vocabulary - something which I took a stab at in #193.